### PR TITLE
Compile less legacy code when BUILD_CAFFE2 is set to False

### PR DESCRIPTION
**Pull request by malfet**

### CODE REVIEW

File: caffe2/CMakeLists.txt [hunk 1]
```diff
@@ -147,6 +149,9 @@ if(NOT INTERN_BUILD_MOBILE OR BUILD_CAFFE2_MOBILE)
   # add_subdirectory(test) # todo: use caffe2_gtest_main instead of gtest_main because we will need to call GlobalInit
   add_subdirectory(transforms)
 endif()
+if(NOT BUILD_CAFFE2)
+  add_subdirectory(proto)
```
[the remark this answers is not in the document]
Comment (malfet) [reply caffe2/CMakeLists.txt:153]: This link that you've shared is broken, but I prefer to keep both cases of `add_subdirectory(proto)` close to each other.
A good followup PR would be to remove the use of proto altogether when not building with caffe2 (but that would require some work on legacy serialization/deserialization code)

**Comment by walterddr** [comment]
adding `ci:all` tag to trigger pure_torch build

**Comment by malfet** [comment]
@walterddr `ci/all` wouldn't work for PRs that originates from the forked repo :(

**Review by walterddr — APPROVED** [review]
lgtm

**Comment by wanchaol** [comment]
this seems triggers the ci failed on the pure_torch build, any ideas how to fix it? or we should revert?

**Comment by walterddr** [comment]
@malfet this seems to be the ATEN test failure I saw earlier - https://app.circleci.com/pipelines/github/pytorch/pytorch/210576/workflows/4f8cef71-eb6b-41e0-97c0-b4913fe048c8/jobs/7272162. I think we should probably rollback for now?

**Pull request closed without merging**